### PR TITLE
1.1 CommandLineParams

### PR DESCRIPTION
**Pull request by Dafinia**

### CODE REVIEW

**Comment by Dafinia** [comment]
@AlexanderSoroka I think it's ready for review, for real this time.

File: JavaHomeworks/1.1 CommandLineParams/CommandLineParams.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+public class CommandLineParams {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.1 CommandLineParams/CommandLineParams.java]: Missed comments for class and method main

File: JavaHomeworks/1.1 CommandLineParams/CommandLineParams.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  public static void main(String[] args) {
+    for (int i = args.length - 1; i >= 0; i--) {
+      System.out.println("Аргумент " + i + " = " + args[i]);
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.1 CommandLineParams/CommandLineParams.java]: Do  not use russian literals

File: JavaHomeworks/1.2 HelloName/HelloName.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Prints out "Hello, Name". Takes a first command line argument (if it exists) and
+ * interprets it as a name. All subsequent command line arguments are ignored.
+ */
+public class HelloName {
+  public static void main(String[] args) {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.2 HelloName/HelloName.java]: Missed comment for main

File: JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/**
+ * A simple calculator that takes 2 arguments from command line and prints out their sum, difference,
+ * product and quotient.
+ */
+public class SimpleCalculator {
+  public static void main(String[] args) {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java]: Missed comment for main

File: JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    double value1;
+    double value2;
+
+    // checking if number of arguments is ok
+    if (args.length != 2) {
+      System.out.println("Invalid number of arguments.");
+      return;
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java]: You should return form main with call System.exit(1); Here 1 is error code that is returned to OS

File: JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    // checking if parsable to double
+    try {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java]: This code is used twice in one method - it's better to make separate method

File: JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      return;
+    }
+    try {
+      value2 = Double.parseDouble(args[1]);
+    } catch (NumberFormatException e) {
+      System.out.println("Second argument is not a number");
+      return;
+    }
+    //checking if arguments are less than max and bigger than min
+    if ((value1 > Double.MAX_VALUE) || (value2 > Double.MAX_VALUE) || (value1 < Double.MIN_VALUE)
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.3 SimpleCalculator/SimpleCalculator.java]: I don't think you can convert value bigger then  Double.MAX_VALUE from String to double without exception, so this check is potentially useless

File: JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
+/**
+ * Quadratic equation solver. Takes 3 arguments from command line and treats them as coefficients a,
+ * b and c of the equation ax^2 + bx +c = 0 and then solves for x.
+ */
+public class QuadraticEquationRoots {
+  public static void main(String[] args) {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java]: Missed comment for main

File: JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    // since it's pure math I think it should be ok to use a, b, c as variable names
+    int a;
+    int b;
+    int c;
+    double discriminant;
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java]: local variable should be declared as close to first use as possible

File: JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      System.out.println("Equation is not quadratic");
+      return;
+    }
+    // calculating discriminant
+    discriminant = b * b - 4 * a * c;
+
+    // just going through all the possible cases, d<0, d==0, d>0
+    if (discriminant < 0) {
+      System.out.println("The discriminant is less than zero. This equation has no real roots");
+    } else if (discriminant == 0) {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java]: Do not use == to compare doubles, take a look to http://www.java67.com/2015/09/float-and-double-value-comparison-in-java-use-relational.html

File: JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    double x2;
+
+    // checking if number of arguments is ok
+    if (args.length != 3) {
+      System.out.println("Invalid number of arguments.");
+      return;
+    }
+    // checking if parsable to int
+    try {
+      a = Integer.parseInt(args[0]);
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java]: Does it mean that user cannot enter double coeffs?

File: JavaHomeworks/1.5 GlenfordMyersTriangle/GlenfordMyersTriangle.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Given lengths of the three sides of a triangle as command line arguments determines whether the
+ * triangle is equilateral, isosceles or scalene.
+ */
+public class GlenfordMyersTriangle {
+  public static void main(String[] args) {
```
Comment (AlexanderSoroka) [inline JavaHomeworks/1.5 GlenfordMyersTriangle/GlenfordMyersTriangle.java]: Missed comment for main

File: JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java (hunk 10 above)
Comment (Dafinia) [inline JavaHomeworks/1.4 QuadraticEquationRoots/QuadraticEquationRoots.java]: Should I use the compare method or what? The article didn't specify what should be used if I really need to check for equality, I don't really see a way around it using relational operators.

**Pull request closed without merging**